pkg/avatar/avatar.go

```diff
@@ -30,10 +33,199 @@ func (p *Player) Type() targets.TargettableType { return targets.TargettablePlay
+	}
+
+	base := atk.Info.Mult*a + atk.Info.FlatDmg
+	damage := base * (1 + dmgBonus)
+
+	//make sure 0 <= cr <= 1
+	if atk.Snapshot.Stats[attributes.CR] < 0 {
+		atk.Snapshot.Stats[attributes.CR] = 0
+	}
+	if atk.Snapshot.Stats[attributes.CR] > 1 {
+		atk.Snapshot.Stats[attributes.CR] = 1
+	}
+
+	char := t.Core.Player.ActiveChar()
+	// Players don't have resistances right now
+	// res := char.Stat(attributes.DendroRes
+	res := 0.0
+
+	def := char.Base.Def*(1+char.Stat(attributes.DEFP)) + char.Stat(attributes.DEF)
+
+	def *= (1 - atk.Info.IgnoreDefPercent)
+	defmod := 1 - def/(def+float64(5*atk.Snapshot.CharLvl)+500)
+
+	//apply def mod
+	damage = damage * defmod
+	//apply resist mod
+
+	resmod := 1 - res/2
+	if res >= 0 && res < 0.75 {
+		resmod = 1 - res
+	} else if res > 0.75 {
+		resmod = 1 / (4*res + 1)
+	}
+	damage = damage * resmod
+
+	precritdmg := damage
+
+	//check if crit
+	if atk.Info.HitWeakPoint || t.Core.Rand.Float64() <= atk.Snapshot.Stats[attributes.CR] {
+		damage = damage * (1 + atk.Snapshot.Stats[attributes.CD])
+		isCrit = true
+	}
+
+	preampdmg := damage
+
+	//calculate em bonus
+	em := atk.Snapshot.Stats[attributes.EM]
+	emBonus := (2.78 * em) / (1400 + em)
+	var reactBonus float64
+	//check melt/vape
+	if atk.Info.Amped {
+		reactBonus = t.Core.Player.ByIndex(atk.Info.ActorIndex).ReactBonus(atk.Info)
+		// t.Core.Log.Debugw("debug", "frame", t.Core.F, core.LogPreDamageMod, "char", t.Index, "char_react", char.CharIndex(), "reactbonus", char.ReactBonus(atk.Info), "damage_pre", damage)
+		damage = damage * (atk.Info.AmpMult * (1 + emBonus + reactBonus))
+	}
+
+	//reduce damage by damage group
+	x := 1.0
+	if !atk.Info.SourceIsSim {
+		x = t.GroupTagDamageMult(atk.Info.ICDTag, atk.Info.ICDGroup, atk.Info.ActorIndex)
+		damage = damage * x
+	}
+
+	if t.Core.Flags.LogDebug {
+		t.Core.Log.NewEvent(
+			atk.Info.Abil,
+			glog.LogCalc,
+			atk.Info.ActorIndex,
+		).
+			Write("src_frame", atk.SourceFrame).
+			Write("damage_grp_mult", x).
+			Write("damage", damage).
+			Write("abil", atk.Info.Abil).
+			Write("talent", atk.Info.Mult).
+			Write("base_atk", atk.Snapshot.BaseAtk).
+			Write("flat_atk", atk.Snapshot.Stats[attributes.ATK]).
+			Write("atk_per", atk.Snapshot.Stats[attributes.ATKP]).
+			Write("use_def", atk.Info.UseDef).
+			Write("base_def", atk.Snapshot.BaseDef).
+			Write("flat_def", atk.Snapshot.Stats[attributes.DEF]).
+			Write("def_per", atk.Snapshot.Stats[attributes.DEFP]).
+			Write("base_hp", atk.Snapshot.BaseHP).
+			Write("flat_hp", atk.Snapshot.Stats[attributes.HP]).
+			Write("hp_per", atk.Snapshot.Stats[attributes.HPP]).
+			Write("total_hp", totalhp).
+			Write("catalyzed", atk.Info.Catalyzed).
+			Write("flat_dmg", atk.Info.FlatDmg).
+			Write("total_atk_def", a).
+			Write("base_dmg", base).
+			Write("ele", st).
+			Write("ele_per", elePer).
+			Write("bonus_dmg", dmgBonus).
+			Write("ignore_def", atk.Info.IgnoreDefPercent).
```

Comment:
Please add back the def_adj log right below this line, but set it to 0 with a comment saying that there are no DefMods applied to the player in sim.